pms_pwa/models/pms_folio.py

```diff
@@ -0,0 +1,84 @@
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+def _get_search_domain(search=False, **post):
+    domains = []
+    search_exists = False
+    if search:
+        search_exists = True
+        for srch in search.split(" "):
+            subdomains = [
+                [("reservation_ids.localizator", "in", [srch])],
+                [("reservation_ids.partner_id.phone", "ilike", srch)],
+                [("reservation_ids.partner_id.mobile", "ilike", srch)],
+                [("reservation_ids.partner_id.name", "ilike", srch)],
+                [("reservation_ids.partner_id.vat", "ilike", srch)],
+                [("reservation_ids.partner_id.email", "ilike", srch)],
+            ]
+            domains.append(expression.OR(subdomains))
+    # REVIEW: Use lib to this
+    # post send a filters with key: "operator&field"
+    # to build a odoo domain:
+    domain_fields = []
+    for k, v in post.items():
+        if v:
+            if k in ["name", "vat", "email"]:
+                domain_fields.append(("reservation_ids.partner_id." + k, "=", v))
+            elif k in ["checkin", "checkout"]:
+                domain_fields.append(("reservation_ids." + k, "=", v))
+            elif k == "checkin_from":
+                domain_fields.append(("reservation_ids.checkin", ">=", v))
+            elif k == "checkout_from":
+                domain_fields.append(("reservation_ids.checkout", ">=", v))
+            elif k == "checkin_to":
+                domain_fields.append(("reservation_ids.checkin", "<=", v))
+            elif k == "checkout_to":
+                domain_fields.append(("reservation_ids.checkout", "<=", v))
+            elif k == "created_from":
+                domain_fields.append(("create_date", ">=", v))
```

Comment:
The user's intention is search by reservation, I think that create, etc should also search with reservation_ids (agency_id and channel_type_id is not necesary beacouse are related fields)

Reply:
Do you mean additional domain conditions for reservations? Something like "reservation_create_from", "reservation_modified_from", "reservation_create_to", etc. ?
I dont know what you mean with agency_id & channel_type_id. In the advanced filters screen you can add filters by origin (agency/channel_type).

Reply:
ok, according to our last conversation, I change the searches by date of creation and modification of the folio by the dates of the reservation. The agency fields and type of sales channel are already made according to the folio, therefore I keep them that way.